x-pack/plugins/enterprise_search/public/applications/app_search/components/source_engines/source_engines_logic.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { kea, MakeLogicType } from 'kea';
+
+import { Meta } from '../../../../../common/types';
+
+import { flashAPIErrors } from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { EngineLogic } from '../engine';
+import { EngineDetails } from '../engine/types';
+
+interface EnginesAPIResponse {
+  results: EngineDetails[];
+  meta: Meta;
+}
+
+interface SourceEnginesLogicValues {
+  dataLoading: boolean;
+  sourceEngines: EngineDetails[];
+}
+
+interface SourceEngineLogicActions {
+  fetchSourceEngines: () => void;
+  setSourceEngines: (
+    sourceEngines: SourceEnginesLogicValues['sourceEngines']
+  ) => { sourceEngines: SourceEnginesLogicValues['sourceEngines'] };
+}
+
+export const SourceEnginesLogic = kea<
+  MakeLogicType<SourceEnginesLogicValues, SourceEngineLogicActions>
+>({
+  path: ['enterprise_search', 'app_search', 'curations_logic'],
+  actions: () => ({
+    fetchSourceEngines: true,
+    setSourceEngines: (sourceEngines) => ({ sourceEngines }),
+  }),
+  reducers: () => ({
+    dataLoading: [
+      true,
+      {
+        setSourceEngines: () => false,
+      },
+    ],
+    sourceEngines: [
+      [],
+      {
+        setSourceEngines: (_, { sourceEngines }) => sourceEngines,
+      },
+    ],
+  }),
+  listeners: ({ actions }) => ({
+    fetchSourceEngines: () => {
```

Comment:
[Optional/high level ask - feel free to punt for another day or a later refactor]

This is basically identical to `MetaEnginesTableLogic`'s fetchSourceEngines, right? Just asking this to get the gears turning, but do you think there's any logical/sensible way to share / combine the two logic files since they feel like they have a lot in common? (I could definitely be wrong about that!)

Maybe one option could be a single shared MetaEnginesBaseLogic file that the two different views extend via Kea's [logic.extend()](https://kea.js.org/docs/guide/advanced#extending-logic) API? 🤷

Reply:
I was thinking about this and it got hard to wrap my head around, I think it's a good idea though, I'd probably refactor this later

Reply:
Sweet!!